### PR TITLE
call onreadystatechange callbacks after change ready-state

### PR DESCRIPTION
**Pull request by senaev**
solves issue https://github.com/sinonjs/nise/issues/60

### CODE REVIEW

File: lib/fake-xhr/index.test.js [hunk 1]
```diff
@@ -1923,7 +1925,9 @@ describe("FakeXMLHttpRequest", function () {
                 readyStateCb;
             var workingXHROverride = function () {
                 this.addEventListener = function (str, fn) {
-                    readyStateCb = fn;
+                    if (str === "readystatechange") {
+                        readyStateCb = fn;
+                    }
                 };
```
Comment (senaev) [inline lib/fake-xhr/index.test.js]: this addEventListener is called with all the listeners in row and before the last listener was readyStateCb, but now is is the last, so we should filter

**Comment by senaev** [comment]
ping

**Comment by senaev** [comment]
I added some improvements to support right readystatechange callback call:
1) change  all fakeRequest properties
2) call other readystatechange listeners
3) call fakeXhr.readyStateCallbacks callback

**Comment by mroderick** [comment]
Thank you for your pull request 🚀 
This has been published to NPM as `nise@1.4.3`

**Pull request merged**